### PR TITLE
build: README: add minikube instructions

### PR DESCRIPTION
**Pull request by philips**
We should use our own tools! Recommend minikube for the VM builds.

cc @dlorenc @r2d4

### CODE REVIEW

**Comment by cblecker** [comment]
@k8s-bot pull-kubernetes-cross test this
@k8s-bot aws test this

**Review by r2d4 — APPROVED** [review]
LGTM

**Comment by spxtr** [comment]
/cc @r2d4 

(IGNORE ME PLEASE)

**Comment by philips** [comment]
@k8s-bot pull-kubernetes-cross test this

File: build/README.md [hunk 1]
```diff
@@ -43,6 +44,17 @@ There are 3 different containers instances that are run from this image.  The fi
 
 All Docker names are suffixed with a hash derived from the file path (to allow concurrent usage on things like CI machines) and a version number.  When the version number changes all state is cleared and clean build is started.  This allows the build infrastructure to be changed and signal to CI systems that old artifacts need to be deleted.
 
+## Minikube
+
+To use minikube you need to create an SSH tunnel for rsync, do something like this:
```
Comment (MHBauer) [inline build/README.md]: I'd avoid the ambiguous language after the comma. Just end the sentence there.

File: build/README.md [hunk 2]
```diff
@@ -43,6 +44,17 @@ There are 3 different containers instances that are run from this image.  The fi
...
+
+```
+export KUBE_RSYNC_PORT=8730
+eval $(minikube docker-env)
+ssh -i ~/.minikube/machines/minikube/id_rsa  $(minikube ip) -L ${KUBE_RSYNC_PORT}:localhost:${KUBE_RSYNC_PORT} -N -l docker &
```
Comment (MHBauer) [inline build/README.md]: ssh into  minikube with minikube creds, start a tunnel into out of the vm with the rsync port, don't do anything, and login as docker. background to keep the tunnel open and let me have my shell back.

I don't think we should advocate allocating sockets in the background so people forget about them. At least not without telling them how to clean up.

File: build/README.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ Building Kubernetes is easy if you take advantage of the containerized build env
 ## Requirements
 
 1. Docker, using one of the following configurations:
+  1. **Minikube on Mac/Windows/Linux** you can use [minikube](https://github.com/kubernetes/minikube) to get a Docker build endpoint. Ensure you have at least 3GB of memory. See instructions for [setting up minikube for a build](#minikube).
```
Comment (MHBauer) [inline build/README.md]: word wrap vs one giant line. I know the rest of the doc isn't like this, but let's clean up gradually.

minikube defaults to 2g memory. I know there's a warning here, but we should probably either set the config to use 3, or explicitly talk about the `--memory` flag on `minikube start`.

**Review by MHBauer — COMMENTED** [review]
I totally agree with the sentiment that we should be using our own tools as much as possible!

File: build/README.md (hunk 3 above)
Comment (philips) [inline build/README.md]: wordwrap to how many chars?

File: build/README.md (hunk 1 above)
Comment (philips) [inline build/README.md]: k

**Comment by philips** [comment]
@k8s-bot pull-kubernetes-cross test this

**Comment by r2d4** [comment]
@aaron-prindle added the `ssh-key` command at head.  We haven't released a version with it yet however.

https://github.com/kubernetes/minikube/pull/1493

File: build/README.md (hunk 2 above)
Comment (philips) [inline build/README.md]: the other instructions in this file do something similar, I get it though.

File: build/README.md (hunk 3 above)
Comment (warmchang) [inline build/README.md]: The default memory is 2G:
"DefaultMemory       = 2048" in file https://github.com/kubernetes/minikube/blob/master/pkg/minikube/constants/constants.go

**Comment by ixdy** [comment]
For dockerized build, what's the advantage of using minikube over just using docker directly?

**Comment by philips** [comment]
The advantage is just fewer virtual machines on your computer. Minikube
provides the docker engine endpoint too so can be used for both.

**Pull request closed without merging**